### PR TITLE
feat(split): use abstractions for flux and prometheus

### PR DESCRIPTION
**Pull request by haarchri**
### Description of your changes
- bump uptest
- enable renovate
- remove service composition for flux, prometheus
- add configuration for xoss and gitops-flux (use abstractions)
- add yamllint
- bump providers

Fixes #

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

```
--- PASS: kuttl (1587.36s)
    --- PASS: kuttl/harness (0.00s)
        --- PASS: kuttl/harness/case (1586.87s)
PASS
10:22:51 [ OK ] running automated tests
```

### CODE REVIEW

File: apis/cluster/composition.yaml [hunk 1]
```diff
@@ -100,7 +129,7 @@ spec:
         kind: Usage
         spec:
           of:
-            apiVersion: aws.platform.upbound.io/v1alpha1
+            apiVersion: azure.platformref.upbound.io/v1alpha1
```
Comment (ytsarev) [inline apis/cluster/composition.yaml:132]: khm, why/how did it work before with this bug? :D

**Review by ytsarev — APPROVED** [review]

File: apis/cluster/composition.yaml (hunk 1 above)
Comment (haarchri) [reply apis/cluster/composition.yaml:132]: this usage is optional - so we didn't catch before

Comment (ytsarev) [reply apis/cluster/composition.yaml:132]: yeah, it's still interesting that deletion ordering worked without it, apparently by accident :)

**Pull request merged**